src/builders.ts

```diff
@@ -166,6 +169,10 @@ export class BomBuilder {
       }
     }
 
+    for (const workspace of this.workspaces) {
+      args.push(`--workspace=${workspace}`)
+    }
```

Comment:
when enabling workspaces, we would also need an option to enable workspace root.

Reply:
I'm not sure I 100% agree/understand the point of this flag from `npm ls`. If I understand correctly, the current behaviour of this plugin when run in a root project is that all dependencies (which includes all workspace dependencies) are included in the output.

The feature requested was to be able to provide a specific workspace when running the plugin in the root so that only dependencies for that specific workspace is included in the the output.

The docs describes the workspace root option as:

```
[include-workspace-root](https://docs.npmjs.com/cli/v10/commands/npm-ls#include-workspace-root)
Default: false
Type: Boolean
Include the workspace root when workspaces are enabled for a command.

When false, specifying individual workspaces via the workspace config, or all workspaces via the workspaces flag, will cause npm to operate only on the specified workspaces, and not on the root project.
```

Does this not mean that if we set this property to true we go back to the original logic (no workspaces provided, use all dependencies found in root project) and if it is set to false (default) then we get our expected output meaning we don't need it?

This is also valid for the `workspaces` command which applies the logic to all workspaces (effectively the same as specifying no workspaces). I'm not sure if these options are not just copied over from other cli commands (such as `install`) and don't make sense for `ls` or if I am misunderstanding something.

Reply:
Hi @jkowalleck, any update regarding my comments above?

Reply:
need to experiment with `npm ls` a bit more. 
added additional demo/setups via https://github.com/CycloneDX/cyclonedx-node-npm/pull/1215 to do so 

will come back later, when i have a deeper insight.

Reply:
> when enabling workspaces, we would also need an option to enable workspace root.

after experimenting with `npm ls --include-workspace-root -w ...`, if found it not having any effect on the results.
 so we do not need it.

all good.